orderbook/src/fee.rs

```diff
@@ -84,6 +84,12 @@ impl MinFeeCalculator {
         amount: Option<U256>,
         kind: Option<OrderKind>,
     ) -> Result<Option<Measurement>> {
+        let buy_token = if buy_token == Some(BUY_ETH_ADDRESS) {
+            Some(self.native_token)
+        } else {
+            buy_token
+        };
+
```

Comment:
I think we need the same logic inside `PriceEstimating` (maybe even only there is fine as I believe MinFeeCalculating calls into it)